pkg/controller/machine/machine_controller.go

```
@@ -234,3 +258,57 @@ func (r *ReconcileMachine) deleteNode(ctx context.Context, cluster *clusterv1.Cl
 
 	return corev1Remote.Nodes().Delete(name, &metav1.DeleteOptions{})
 }
+
+func (r *ReconcileMachine) drainNode(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	var kubeClient kubernetes.Interface
+	if cluster == nil {
+		var err error
+		kubeClient, err = kubernetes.NewForConfig(r.config)
+		if err != nil {
+			return fmt.Errorf("unable to build kube client: %v", err)
+		}
+	} else {
+		// Otherwise, proceed to get the remote cluster client and get the Node.
+		remoteClient, err := remote.NewClusterClient(r.Client, cluster)
+		if err != nil {
+			klog.Errorf("Error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
+				cluster.Name, machine.Name, err)
+			return nil
+		}
+		var err2 error
+		kubeClient, err2 = kubernetes.NewForConfig(remoteClient.RESTConfig())
+		if err2 != nil {
+			klog.Errorf("Error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
+				cluster.Name, machine.Name, err)
+			return nil
+		}
+	}
+
+	node, err := kubeClient.CoreV1().Nodes().Get(machine.Status.NodeRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to get node %q: %v", machine.Status.NodeRef.Name, err)
+	}
+
+	if err := kubedrain.Drain(
+		kubeClient,
+		[]*corev1.Node{node},
+		&kubedrain.DrainOptions{
+			Force:              true,
+			IgnoreDaemonsets:   true,
+			DeleteLocalData:    true,
+			GracePeriodSeconds: -1,
+			Logger:             info.New(klog.V(0)),
+			// If a pod is not evicted in 20 second, retry the eviction next time the
+			// machine gets reconciled again (to allow other machines to be reconciled)
+			Timeout: 20 * time.Second,
+		},
+	); err != nil {
+		// Machine still tries to terminate after drain failure
+		klog.Warningf("drain failed for machine %q: %v", machine.Name, err)
+		return &controllerError.RequeueAfterError{RequeueAfter: 20 * time.Second}
```

Why not just return the error? (Requeue happens automatically)